src/test/java/teammates/test/cases/ui/browsertests/InstructorCourseEditPageUiTest.java

```diff
@@ -178,44 +175,98 @@ private void testEditInstructorAction() {
+        assertTrue(courseEditPage.isPrivilegeCheckboxInModalChecked("canviewsessioninsection"));
+        assertTrue(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifysessioncommentinsection"));
+        courseEditPage.closeModal();
+        
+        // Observer
+        courseEditPage.clickViewDetailsLinkForInstructor(instructorIndex, 3);
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifycourse"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifyinstructor"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifysession"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifystudent"));
+        assertTrue(courseEditPage.isPrivilegeCheckboxInModalChecked("canviewstudentinsection"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("cangivecommentinsection"));
+        assertTrue(courseEditPage.isPrivilegeCheckboxInModalChecked("canviewcommentinsection"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifycommentinsection"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("cansubmitsessioninsection"));
+        assertTrue(courseEditPage.isPrivilegeCheckboxInModalChecked("canviewsessioninsection"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifysessioncommentinsection"));
+        courseEditPage.closeModal();
+        
+        // Tutor
+        courseEditPage.clickViewDetailsLinkForInstructor(instructorIndex, 4);
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifycourse"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifyinstructor"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifysession"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifystudent"));
+        assertTrue(courseEditPage.isPrivilegeCheckboxInModalChecked("canviewstudentinsection"));
+        assertTrue(courseEditPage.isPrivilegeCheckboxInModalChecked("cangivecommentinsection"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canviewcommentinsection"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifycommentinsection"));
+        assertTrue(courseEditPage.isPrivilegeCheckboxInModalChecked("cansubmitsessioninsection"));
+        assertTrue(courseEditPage.isPrivilegeCheckboxInModalChecked("canviewsessioninsection"));
+        assertFalse(courseEditPage.isPrivilegeCheckboxInModalChecked("canmodifysessioncommentinsection"));
+        courseEditPage.closeModal();
         
-        assertEquals(true, courseEditPage.clickEditInstructorLink(1));
         // this should be click co-owner role
-        courseEditPage.clickViewDetailsLinkForInstructor(1, 1);
+        courseEditPage.clickViewDetailsLinkForInstructor(instructorIndex, 1);
         // what for the animation to finish
```

Comment:
This is not relevant to this pull request and not added by you, but perhaps you can remove these comments if they are not useful anymore